idaes/core/util/parameter_sweep.py

```diff
@@ -428,30 +442,36 @@ def execute_single_sample(self, sample_id: int):
         Returns:
             results: results generated by collect_results callback
             solved: bool indicating whether solver reported optimal convergence
+            error: str if error occurs during solve else None
         """
         model = self.get_initialized_model()
 
         # Load sample values
         self.set_input_values(model, sample_id)
 
-        # Solve model and capture output
-        solver = self._get_solver()
+        if self.config.solver is None:
+            raise ConfigurationError("Please specify a solver to use.")
```

Comment:
Should `solver` be required? You could imagine somebody implementing a `run_model` that doesn't use the second argument. Maybe it just computes condition number or something.

Reply:
I removed this check here, and just pass the value on to the `run_model` callback to use as required (or not). Due to this, I also changed what the `run_model` method returns as before we expected a Pyomo results object back. Instead I now expect a bool indicating success or failure and a general purpose "run_stats" which can contain any information the user wants to return for collection (by default this is the results object).